src/inmanta/data/__init__.py

```diff
@@ -2523,6 +2524,20 @@ def to_dto(self) -> m.Environment:
             validator=convert_int,
             agent_restart=True,
         ),
+        AUTOSTART_AGENT_RESTORE_STATE: Setting(
+            name=AUTOSTART_AGENT_RESTORE_STATE,
+            typ="bool",
+            default=True,
+            doc=(
+                "Restore state of the resource scheduler when the agent (re)starts to continue (close to) where it left off."
+                " This is the recommended behavior as it results in more efficient deploy scheduling at startup."
+                " The alternative is to discard all but the most basic resource state, and to fall back to deploy increment"
+                " calculation as was the norm in older versions of the Inmanta orchestrator (<= iso7/2024)."
+                " We recommend to disable this setting only in the unlikely event of certain state inconsistencies in order"
+                " to start the agent with a clean slate."
```

Comment:
Fast restart allows the scheduler to restart faster. When disabled, the restart is more robust. Disabling this option can help to resolve scheduler problems.